storage/inmem/inmem.go

```diff
@@ -327,11 +345,45 @@ func (h *handle) Unregister(_ context.Context, txn storage.Transaction) {
 }
 
 func (db *store) runOnCommitTriggers(ctx context.Context, txn storage.Transaction, event storage.TriggerEvent) {
+	if db.returnASTValuesOnRead && len(db.triggers) > 0 {
+		// FIXME: Not very performant for large data.
```

Comment:
Maybe this a caveat we should point out about using this feature so users are aware of this. There seems to be no way to escape the conversion here.

Reply:
Yeah .. an alternative would be to not do the conversion here, and push that responsibility to the consumer. Then conversion would only happen when the consumer is interested in the value. Or the consumer could even be updated to deal with AST values in addition to raw data. That'd be a breach in interface contract, though, and a pretty gnarly breaking change that is only detected at runtime, not compiletime.

Reply:
Imo making the consequences more clear seems better that making a breaking change in this case.

Reply:
We can always optimize.